python/ray/data/read_api.py

```diff
@@ -2122,42 +2094,27 @@ def from_huggingface(
             )
 
     Args:
-        dataset: A `Hugging Face Datasets Dataset`_ or `DatasetDict`_.
-            :class:`~ray.data.IterableDataset` isn't supported.
+        dataset: A `Hugging Face Datasets Dataset`_.
+            :class:`~ray.data.IterableDataset` and
+            `DatasetDict <https://huggingface.co/docs/datasets/package_reference/main_classes#datasets.DatasetDict/>`_
+            are not supported.
 
     Returns:
-        A :class:`~ray.data.Dataset` holding rows from the `Hugging Face Datasets Dataset`_,
-        or a dict of :class:`Datasets <ray.data.Dataset>` in case ``dataset`` is a `DatasetDict`_.
+        A :class:`~ray.data.Dataset` holding rows from the `Hugging Face Datasets Dataset`_.
     """  # noqa: E501
     import datasets
 
-    def convert(ds: "datasets.Dataset") -> Dataset:
+    if isinstance(dataset, datasets.Dataset):
         # To get the resulting Arrow table from a Hugging Face Dataset after
         # applying transformations (e.g. train_test_split(), shard(), select()),
         # we create a copy of the Arrow table, which applies the indices
         # mapping from the transformations.
-        hf_ds_arrow = ds.with_format("arrow")
+        hf_ds_arrow = dataset.with_format("arrow")
         ray_ds = from_arrow(hf_ds_arrow[:])
         return ray_ds
-
-    if isinstance(dataset, datasets.DatasetDict):
-        available_keys = list(dataset.keys())
-        logger.warning(
-            "You provided a Huggingface DatasetDict which contains multiple "
```

Comment:
Let's keep this, but change to an error message

Reply:
would the last case raising `TypeError` cover this case as well? or do we want to explicitly raise a separate error for this case, with a message like "DatasetDict is not supported"?

Reply:
Btw, it should be a DeprecationWarning raised since technically this is a breaking change right.